tensordict/functional.py

```diff
@@ -163,7 +177,7 @@ def pad_sequence(
     for i, td in enumerate(list_of_tensordicts):
 
         for key in keys:
-            item = td.get(key)
+            item = td._tensordict.get(key) if is_tensorclass(td) else td.get(key)
```

Comment:
is_tensorclass is a bit too expensive, let's take this check out of the loop (isolate the tensordict we'll be using before iterating through the keys)